apps/src/code-studio/components/ShareAllowedDialog.jsx

```diff
@@ -327,6 +338,12 @@ class ShareAllowedDialog extends React.Component {
                   />
                   }
                   {/* prevent buttons from overlapping when unpublish is pending */}
+                  {canPrint && hasThumbnail &&
+                    <a href="#" onClick={this.print}>
+                      <i className="fa fa-print" style={{fontSize: 26}} />
+                      <span>{i18n.print()}</span>
+                    </a>
+                  }
```

Comment:
nit: this new code should go just before the "prevent buttons" comment, unless that comment applies to this new code in some way